php/Block.php

```diff
@@ -63,66 +63,213 @@ public function register_block() {
 	 * @return string The markup of the block.
 	 */
 	public function render_callback( $attributes, $content, $block ) {
-		$post_types = get_post_types(  [ 'public' => true ] );
-		$class_name = $attributes['className'];
-		ob_start();
 
-		?>
-        <div class="<?php echo $class_name; ?>">
-			<h2>Post Counts</h2>
-			<ul>
-			<?php
-			foreach ( $post_types as $post_type_slug ) :
-                $post_type_object = get_post_type_object( $post_type_slug  );
-                $post_count = count(
-                    get_posts(
+		// Store this in a local var for caching since we'll reference it a few
+		// times in this method.
+		$current_post_id = get_the_ID();
+
+		/**
+		 * Fetch all public post types as objects, and construct a new array that
+		 * maps the post type's label name (not slug), and the number of
+		 * published posts within that post type so we can easily display it
+		 * below.
+		 */
+
+		// Array to store our new values in.
+		$public_post_type_counts = [];
+
+		// Get all public post type objects.
+		$post_type_objects = get_post_types(  [ 'public' => true ], 'objects' );
+
+		// Validate the values.
+		if (
+			! empty( $post_type_objects )
+			&& is_array( $post_type_objects )
+		) {
+
+			// Loop through each result and store the label and count mapped to
+			// the slug.
+			foreach ( $post_type_objects as $post_type_object ) {
+				/**
+				 * @var array $public_post_type_counts = [
+				 *     'posts' => [
+				 *         'label' => 'Posts',
+				 *         'count' => 5,
+				 *      ],
+				 *      'pages' => [
+				 *        'label' => 'Pages',
+				 *        'count' => 3,
+				 *      ],
+				 * ]
+				 */
+				$public_post_type_counts[ $post_type_object->name ] = [
+					'label' => $post_type_object->labels->name ?? '',
+					'count' => wp_count_posts( $post_type_object->name )->publish ?? 0,
```

Comment:
The `attachment` post type uses `inherit` instead of `publish` as the default post status, meaning it will always print `There are 0 Media.`.

I did not know if this is a bug, or a feature.

I'd either remove `attachment` from the list (narrowing down what we consider to be "public"), or I'd update the `publish` to include a `inherit` conditional. I'd need context and direction on that decision.